### PR TITLE
Allow exporting a binding before its declaration

### LINKED ISSUE #14679: Use before declaration false positive when combined with binding exports
**TypeScript Version:**  2.2.1

**Code**

```ts
import methodFactory from './http-client'

export { httpDelete as delete } // block scoped httpDelete used before declaration
const httpDelete = methodFactory('DELETE')
```

**Expected behavior:**

No error. The export declaration makes no use of the variable; what should happen is that `httpDelete`'s TDZ should extend to the `delete` export, but that happens regardless of the relative position of the `export` declaration.

**Actual behavior:**

Two errors in the same line about the variable being used before assignment.

This error is correct in the case of `export default` and the typescript-specific `export =`, but not in any other use of `export`.

### PR DESCRIPTION
**Pull request by Kovensky**
Fixes #14679

### CODE REVIEW

File: tests/cases/conformance/es6/modules/exportBinding.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export { x }
+const x = 'x'
+
+export { Y as Z }
```
Comment (Kovensky) [inline tests/cases/conformance/es6/modules/exportBinding.ts]: Strangely, this was not an error; only `const` and `let` were errors, even though `class Foo {}` has the exact same behavior as `let Foo = class {}` (no hoisting).

**Comment by RyanCavanaugh** [comment]
LGTM

**Comment by RyanCavanaugh** [comment]
Last call before I merge this

**Comment by sandersn** [comment]
Sorry, looks like we forgot about this one. I'll merge from master and then merge the PR once Travis runs one more time.

**Comment by sandersn** [comment]
OK, I manually merged after following github's instructions, which don't automatically trigger Travis. So...uh...I'm watching Travis build on master right now. 😅

**Pull request merged**